src/output/workspace-impl.cpp

```diff
@@ -230,13 +230,23 @@ struct workspace_set_t::impl
             auto wm  = view->get_geometry();
             float px = 1. * wm.x / old_w;
             float py = 1. * wm.y / old_h;
-            float pw = 1. * wm.width / old_w;
-            float ph = 1. * wm.height / old_h;
 
-            view->set_geometry({
-                    int(px * new_geometry.width), int(py * new_geometry.height),
-                    int(pw * new_geometry.width), int(ph * new_geometry.height)
-                });
+            auto edges = view->toplevel()->current().tiled_edges;
+            if (view->get_output() && view->toplevel()->current().fullscreen)
+            {
+                wf::get_core().default_wm->fullscreen_request(view, view->get_output(), false);
+                wf::get_core().default_wm->fullscreen_request(view, view->get_output(), true);
+            } else if (edges)
+            {
+                wf::get_core().default_wm->tile_request(view, 0);
```

Comment:
Grid already handles this without untiling and re-tiling

https://github.com/WayfireWM/wayfire/blob/7ce8332fe63583933d1a38494178be7d5bafad94/plugins/grid/grid.cpp#L145

In general I feel like un-tiling and re-tiling is a very hacky thing to do, esp. since for example tiled state is set by the simple-tile plugin. You ought to not just untile views like that ...

Reply:
So I think that in general we can just not do anything for this case. Alternatively, we'd need a new weird signal, `adjust_view_on_workspace_set_resize_signal` which then simple-tile/grid can register to, say 'I have adjusted the view' (with a carried_out flag), and if nobody takes care of those views, then core can do that - set fullscreen/tiled geometry or only adjust the position.

Reply:
I guess, we can take the code from grid. Let me see how I can incorporate it.

Reply:
No, the code should remain in grid! Core doesn't know about all the available slots.

Reply:
I see. So, how do we "retile" tiled (maximized) views, then?

Reply:
Nevermind.. I see that they remain tiled, automagically!! :)

Reply:
Not magically, the grid plugin does that :)

The only problem is if the grid plugin is not enabled, but I think this is very very rare in actual Wayfire setups, so I don't consider it a problem that we don't re-maximize if grid is not enabled :)

Reply:
Okay... We now do nothing when the view is maximized!

(I=When I said automagically, I meant that we don't need to add any code here..! At that time I had not yet realised that grid plugin listens to workarea_changed_signal. :smiley: But, thank for the explanation.)